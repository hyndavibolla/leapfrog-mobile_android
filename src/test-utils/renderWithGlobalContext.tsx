import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
