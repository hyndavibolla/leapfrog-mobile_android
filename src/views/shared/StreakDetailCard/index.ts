export { default as StreakDetailCard } from './StreakDetailCard';
