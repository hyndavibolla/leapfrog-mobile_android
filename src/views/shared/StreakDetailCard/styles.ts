import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE, LINE_HEIGHT } from '_constants';

export const styles = {
  content: css`
    border-color: ${COLOR.GRAY};
    border-width: 1px;
    border-radius: 8px;
    margin-top: 8px;
    background-color: ${COLOR.WHITE};
    border-radius: 8px;
    flex-direction: row;
    align-items: center;
    padding: 16px;
  `,
  completeContent: css`
    background-color: ${COLOR.LIGHT_GRAY};
  `,
  streakImageContainer: css`
    background-color: ${COLOR.GRAY};
    border-radius: 40px;
  `,
  streakImage: css`
    width: 40px;
    height: 40px;
  `,
  infoContainer: css`
    flex: 1;
    margin-left: 12px;
  `,
  title: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.SMALL};
    font-family: ${FONT_FAMILY.BOLD};
  `,
  completeTitle: css`
    font-family: ${FONT_FAMILY.BOLD};
    color: ${COLOR.DARK_GRAY};
  `,
  completeSubtitle: css`
    font-family: ${FONT_FAMILY.MEDIUM};
  `,
  progressContainer: css`
    flex-direction: row;
    justify-content: space-between;
  `,
  footerContainer: css`
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    margin-top: 20px;
  `,
  progressText: css`
    color: ${COLOR.DARK_GRAY};
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.PETITE};
    line-height: ${LINE_HEIGHT.SMALL};
  `,
  footerText: css`
    color: ${COLOR.DARK_GRAY};
    font-size: ${FONT_SIZE.PETITE};
    font-family: ${FONT_FAMILY.HEAVY};
    margin-right: 8px;
  `,
  completeText: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-size: ${FONT_SIZE.SMALLER};
  `,
  iconListContainer: css`
    flex-direction: row;
  `,
  icon: css`
    margin-right: 4px;
  `,
  progressTextWarn: css`
    color: ${COLOR.ORANGE};
  `,
  daysDiffContainer: css`
    justify-content: center;
  `
};
