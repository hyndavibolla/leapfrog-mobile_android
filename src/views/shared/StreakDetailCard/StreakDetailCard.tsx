import React, { memo, useEffect, useMemo, useState } from 'react';
import { View } from 'react-native';
import moment from 'moment';

import { Icon } from '_commons/components/atoms/Icon';
import { COLOR, FONT_FAMILY, FONT_SIZE, ICON, LINE_HEIGHT } from '_constants';
import { BrandLogo } from '_components/BrandLogo';
import { Text } from '_components/Text';
import { StreakModel } from '_models';
import { getFormattedTimeDiff } from '_utils/getFormattedTimeDiff';
import { useTestingHelper } from '_utils/useTestingHelper';

import { styles } from './styles';

export interface Props {
  streak: StreakModel.IStreak;
  brandName: string;
}

export default memo(({ streak, brandName }: Props) => {
  const { getTestIdProps } = useTestingHelper('streak-detail-card');
  const [isInProgress, setIsInProgress] = useState<boolean>(false);
  const isComplete = streak.currentQualifiedValue >= streak.thresholdValue;
  const formattedTimeDiff = getFormattedTimeDiff(streak.endDt, Date.now());
  const daysDiff = moment(streak.endDt).diff(Date.now(), 'days');
  const hasEnded = moment(streak.endDt).isSameOrBefore(Date.now());
  const criticalDayThreshold = 5;

  useEffect(() => {
    setIsInProgress(!isComplete && streak.currentQualifiedValue > 0);
  }, [isComplete, streak.currentQualifiedValue]);

  const daysDiffView = useMemo(() => {
    if (isComplete)
      return (
        <Text color={COLOR.PRIMARY_BLUE} font={FONT_FAMILY.BOLD} lineHeight={LINE_HEIGHT.SMALL} size={FONT_SIZE.PETITE} {...getTestIdProps('complete')}>
          Completed!
        </Text>
      );
    if (hasEnded) return null;
    return <Text style={[styles.progressText, daysDiff <= criticalDayThreshold && styles.progressTextWarn]}>{formattedTimeDiff} left</Text>;
  }, [isComplete, hasEnded, daysDiff, criticalDayThreshold, formattedTimeDiff, getTestIdProps]);

  return (
    <>
      <Text color={COLOR.BLACK} font={FONT_FAMILY.BOLD} lineHeight={LINE_HEIGHT.MEDIUM} size={FONT_SIZE.SMALLER}>
        {`${isInProgress ? 'Complete' : 'Activate'} a Mission now ðŸ”¥`}
      </Text>
      <Text color={COLOR.DARK_GRAY} font={FONT_FAMILY.MEDIUM} lineHeight={LINE_HEIGHT.PARAGRAPH} size={FONT_SIZE.PETITE}>
        Buy on {brandName} and earn a {streak.title} Stamp.
      </Text>
      <View style={[styles.content, (isComplete || hasEnded) && styles.completeContent]} {...getTestIdProps('container')}>
        <View style={styles.streakImageContainer}>
          <BrandLogo image={streak.imageUrl} style={styles.streakImage} size={40} />
        </View>
        <View style={styles.infoContainer}>
          <View style={styles.progressContainer}>
            <View style={styles.iconListContainer}>
              {Array(streak.thresholdValue)
                .fill(null)
                .map((_, index) => {
                  const isCurrentIncomplete = index >= streak.currentQualifiedValue;
                  return (
                    <View key={`${index}-${isCurrentIncomplete}`} style={styles.icon}>
                      {isCurrentIncomplete ? (
                        <Icon name={ICON.MISSION_CIRCLE} size={FONT_SIZE.BIG} color={COLOR.DARK_GRAY} />
                      ) : (
                        <Icon name={ICON.MISSION_CIRCLE} size={FONT_SIZE.BIG} />
                      )}
                    </View>
                  );
                })}
            </View>
            <View style={styles.daysDiffContainer}>{daysDiffView}</View>
          </View>
        </View>
      </View>
    </>
  );
});
