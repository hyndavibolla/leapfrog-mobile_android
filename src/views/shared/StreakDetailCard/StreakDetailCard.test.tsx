import React from 'react';

import { getStreak_1 } from '_test_utils/entities';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import StreakDetailCard, { Props } from './StreakDetailCard';

describe('StreakDetailCard', () => {
  Date.now = () => 1980;
  let props: Props;
  beforeEach(() => {
    props = {
      streak: getStreak_1(),
      brandName: 'brand name :D'
    };
  });

  it('should render', () => {
    props.streak.currentQualifiedValue = 1;
    props.streak.thresholdValue = 3;
    const { toJSON, queryByTestId } = renderWithGlobalContext(<StreakDetailCard {...props} />);
    expect(queryByTestId('streak-detail-card-complete')).toBeFalsy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render close to end date', () => {
    props.streak.currentQualifiedValue = 1;
    props.streak.thresholdValue = 3;
    props.streak.endDt = 1980 + 1000 * 60 * 60 * 24 * 2; /** 2 days */
    const { toJSON, queryByTestId } = renderWithGlobalContext(<StreakDetailCard {...props} />);
    expect(queryByTestId('streak-detail-card-complete')).toBeFalsy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render complete', () => {
    props.streak.currentQualifiedValue = 3;
    props.streak.thresholdValue = 3;
    const { toJSON, queryByTestId } = renderWithGlobalContext(<StreakDetailCard {...props} />);
    expect(queryByTestId('streak-detail-card-complete')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with expired streak', () => {
    props.streak.endDt = Date.now();
    props.streak.currentQualifiedValue = 1;
    props.streak.thresholdValue = 3;
    const { toJSON } = renderWithGlobalContext(<StreakDetailCard {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
