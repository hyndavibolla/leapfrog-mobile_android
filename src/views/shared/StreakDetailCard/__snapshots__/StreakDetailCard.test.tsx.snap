// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreakDetailCard should render 1`] = `
Array [
  <Text
    color="#262C2F"
    font="SFProDisplay-Bold"
    lineHeight="24px"
    size="16px"
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Complete a Mission now ðŸ”¥
  </Text>,
  <Text
    color="#909BA0"
    font="SFProDisplay-Medium"
    lineHeight="22px"
    size="14px"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 14,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Buy on 
    brand name :D
     and earn a 
    Grocery & Delivery Streak
     Stamp.
  </Text>,
  <View
    accessibilityLabel="streak-detail-card-container"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#E2E3E3",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#E2E3E3",
          "borderLeftWidth": 1,
          "borderRightColor": "#E2E3E3",
          "borderRightWidth": 1,
          "borderTopColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "marginTop": 8,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        false,
      ]
    }
    testID="streak-detail-card-container"
  >
    <View
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "borderTopLeftRadius": 40,
          "borderTopRightRadius": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "height": 40,
              "width": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontFamily": "SFProDisplay-Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  false,
                ],
              ]
            }
          >
            51 years
             left
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`StreakDetailCard should render close to end date 1`] = `
Array [
  <Text
    color="#262C2F"
    font="SFProDisplay-Bold"
    lineHeight="24px"
    size="16px"
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Complete a Mission now ðŸ”¥
  </Text>,
  <Text
    color="#909BA0"
    font="SFProDisplay-Medium"
    lineHeight="22px"
    size="14px"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 14,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Buy on 
    brand name :D
     and earn a 
    Grocery & Delivery Streak
     Stamp.
  </Text>,
  <View
    accessibilityLabel="streak-detail-card-container"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#E2E3E3",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#E2E3E3",
          "borderLeftWidth": 1,
          "borderRightColor": "#E2E3E3",
          "borderRightWidth": 1,
          "borderTopColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "marginTop": 8,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        false,
      ]
    }
    testID="streak-detail-card-container"
  >
    <View
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "borderTopLeftRadius": 40,
          "borderTopRightRadius": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "height": 40,
              "width": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontFamily": "SFProDisplay-Medium",
                    "fontSize": 14,
                    "lineHeight": 18,
                  },
                  Object {
                    "color": "#FF5F00",
                  },
                ],
              ]
            }
          >
            2 days
             left
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`StreakDetailCard should render complete 1`] = `
Array [
  <Text
    color="#262C2F"
    font="SFProDisplay-Bold"
    lineHeight="24px"
    size="16px"
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Activate a Mission now ðŸ”¥
  </Text>,
  <Text
    color="#909BA0"
    font="SFProDisplay-Medium"
    lineHeight="22px"
    size="14px"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 14,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Buy on 
    brand name :D
     and earn a 
    Grocery & Delivery Streak
     Stamp.
  </Text>,
  <View
    accessibilityLabel="streak-detail-card-container"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#E2E3E3",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#E2E3E3",
          "borderLeftWidth": 1,
          "borderRightColor": "#E2E3E3",
          "borderRightWidth": 1,
          "borderTopColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "marginTop": 8,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        Object {
          "backgroundColor": "#F5F7FA",
        },
      ]
    }
    testID="streak-detail-card-container"
  >
    <View
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "borderTopLeftRadius": 40,
          "borderTopRightRadius": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "height": 40,
              "width": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessibilityLabel="streak-detail-card-complete"
            accessible={true}
            color="#0066CC"
            font="SFProDisplay-Bold"
            lineHeight="18px"
            size="14px"
            style={
              Array [
                Object {
                  "color": "#0066CC",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 14,
                  "lineHeight": 18,
                },
                undefined,
              ]
            }
            testID="streak-detail-card-complete"
          >
            Completed!
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`StreakDetailCard should render with expired streak 1`] = `
Array [
  <Text
    color="#262C2F"
    font="SFProDisplay-Bold"
    lineHeight="24px"
    size="16px"
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
      ]
    }
  >
    Complete a Mission now ðŸ”¥
  </Text>,
  <Text
    color="#909BA0"
    font="SFProDisplay-Medium"
    lineHeight="22px"
    size="14px"
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 14,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Buy on 
    brand name :D
     and earn a 
    Grocery & Delivery Streak
     Stamp.
  </Text>,
  <View
    accessibilityLabel="streak-detail-card-container"
    accessible={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#E2E3E3",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderBottomWidth": 1,
          "borderLeftColor": "#E2E3E3",
          "borderLeftWidth": 1,
          "borderRightColor": "#E2E3E3",
          "borderRightWidth": 1,
          "borderTopColor": "#E2E3E3",
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "borderTopWidth": 1,
          "flexDirection": "row",
          "marginTop": 8,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        Object {
          "backgroundColor": "#F5F7FA",
        },
      ]
    }
    testID="streak-detail-card-container"
  >
    <View
      style={
        Object {
          "backgroundColor": "#E2E3E3",
          "borderBottomLeftRadius": 40,
          "borderBottomRightRadius": 40,
          "borderTopLeftRadius": 40,
          "borderTopRightRadius": 40,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 60,
              "width": 60,
            },
            Object {
              "height": 40,
              "width": 40,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <FastImageView
            accessibilityLabel="brand-logo-image"
            accessible={true}
            onFastImageError={[Function]}
            onFastImageLoad={[Function]}
            onFastImageLoadEnd={[Function]}
            onFastImageLoadStart={[Function]}
            resizeMode="cover"
            source={
              Object {
                "priority": "high",
                "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="brand-logo-image"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 12,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0066CC",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
          <View
            style={
              Object {
                "marginRight": 4,
              }
            }
          >
            <Text
              accessibilityLabel="icon-text"
              accessible={true}
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#909BA0",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "sywmax",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="icon-text"
            >
              î¥†
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>,
]
`;
